test/utils/utils.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import {
+	ByRoleMatcher,
+	ByRoleOptions,
+	GetAllBy,
+	queries,
+	queryHelpers,
+	screen as rtlScreen,
+	within as rtlWithin,
+	type Screen
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { filter } from 'lodash';
+
+export type UserEvent = ReturnType<(typeof userEvent)['setup']>;
+
+type ByRoleWithIconOptions = ByRoleOptions & {
+	icon: string | RegExp;
+};
+/**
+ * Matcher function to search an icon button through the icon data-testid
+ */
+const queryAllByRoleWithIcon: GetAllBy<[ByRoleMatcher, ByRoleWithIconOptions]> = (
+	container,
+	role,
+	{ icon, ...options }
+) =>
+	filter(
+		rtlWithin(container).queryAllByRole(role, options),
+		(element) => rtlWithin(element).queryByTestId(icon) !== null
+	);
+const getByRoleWithIconMultipleError = (
+	container: Element | null,
+	role: ByRoleMatcher,
+	options: ByRoleWithIconOptions
+): string => `Found multiple elements with role ${role} and icon ${options.icon}`;
+const getByRoleWithIconMissingError = (
```

Comment:
I don't understand what is the utility of this function

Reply:
also part of the custom query implementation, please refer to https://github.com/zextras/carbonio-ui-commons/pull/88#discussion_r1439262700